utils/dlinfo_wrapper.c

```diff
@@ -0,0 +1,30 @@
+#define _GNU_SOURCE
+#include <dlfcn.h>
+#include <link.h>
+#include <stdlib.h>
+#include <string.h>
+#include <linux/limits.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+const char *find_lib_path(const char* lib_name, bool verbose) {
+  void* handle = dlopen(lib_name, RTLD_LAZY);
+  if (!handle) goto err0;
+
+  struct link_map* lm = NULL;
+  if (dlinfo(handle, RTLD_DI_LINKMAP, &lm) != 0) goto err1;
+
+  char *lib_path = calloc(PATH_MAX + 1, sizeof(char));
+  strncpy(lib_path, lm->l_name, PATH_MAX);
```

Comment:
```suggestion
  char *lib_path = calloc(strlen(lm->l_name) + 1, sizeof(char));
  strncpy(lib_path, lm->l_name, strlen(lm->l_name) + 1);
```
This should sidestep any PATH_MAX issues.
You may want to check for null value of `lm->l_name` though I don't know if it is really a possibility.

Reply:
The doc does not inspire confidence:

> **Macro: int PATH_MAX**
>    The uniform system limit (if any) for the length of an entire file name (that is, the argument given to system calls such as open), including the terminating null character.
>    Portability Note: The GNU C Library does not enforce this limit even if PATH_MAX is defined.